src/modules/avformat/producer_avformat.c

```diff
@@ -1658,6 +1681,53 @@ static int ignore_send_packet_result(int result)
 	return result >= 0 || result == AVERROR(EAGAIN) || result == AVERROR_EOF || result == AVERROR_INVALIDDATA || result == AVERROR(EINVAL);
 }
 
+static void* packets_worker( void *param )
+{
+	producer_avformat self = param;
+	AVPacket *pkt = av_packet_alloc();
+	if ( !pkt ) {
+		mlt_log_error( MLT_PRODUCER_SERVICE(self->parent), "av_packet_alloc failed\n" );
+		exit( EXIT_FAILURE );
+	}
+
+	for (;;) {
+		pthread_mutex_lock( &self->packets_mutex );
+
+	check_stop:
```

Comment:
I am not sure, but I expect the packets worker will also need to handle seek so that seeking does not occur during reading. Maybe add a "check_seek" step in this loop?

Reply:
packets_mutex is locked prior to seeks, unless there is some other seeking logic I missed.